pennylane/ops/qutrit/matrix_ops.py

```diff
@@ -98,20 +109,214 @@ def compute_matrix(U):  # pylint: disable=arguments-differ
+    both wires ``0`` and ``1``:
+
+    >>> U = np.array([[1, 1, 0], [1, -1, 0], [0, 0, np.sqrt(2)]]) / np.sqrt(2)
+    >>> qml.ControlledQutritUnitary(U, control_wires=[0, 1], wires=2)
+
+    By default, controlled operations apply the desired gate if the control qutrit(s)
+    are all in the state :math:`\vert 2\rangle`. However, there are some situations where
+    it is necessary to apply a gate conditioned on all control qutrits being in the
+    :math:`\vert 0\rangle` or :math:`\vert 1\rangle` state, or a mix of the three.
+
+    The state on which to control can be changed by passing a string of trits to
+    `control_values`. For example, if we want to apply a single-qutrit unitary to
+    wire ``3`` conditioned on three wires where the first is in state ``0``, the
+    second is in state ``1``, and the third in state ``2``, we can write:
+
+    >>> qml.ControlledQutritUnitary(U, control_wires=[0, 1, 2], wires=3, control_values='012')
+    """
+    num_wires = AnyWires
+    """int: Number of wires that the operator acts on."""
+
+    num_params = 1
+    """int: Number of trainable parameters that the operator depends on."""
+
+    ndim_params = (2,)
+    """tuple[int]: Number of dimensions per trainable parameter that the operator depends on."""
+
+    grad_method = None
+    """Gradient computation method."""
+
+    def __init__(
+        self,
+        *params,
+        control_wires=None,
+        wires=None,
+        control_values=None,
+        do_queue=True,
+    ):
+        if control_wires is None:
+            raise ValueError("Must specify control wires")
+
+        wires = Wires(wires)
+        control_wires = Wires(control_wires)
+
+        if Wires.shared_wires([wires, control_wires]):
+            raise ValueError(
+                "The control wires must be different from the wires specified to apply the unitary on."
+            )
+
+        self._hyperparameters = {
+            "u_wires": wires,
+            "control_wires": control_wires,
+            "control_values": control_values,
+        }
+
+        total_wires = control_wires + wires
+        super().__init__(*params, wires=total_wires, do_queue=do_queue)
+
+    @staticmethod
+    def compute_decomposition(*params, wires=None, **hyperparameters):
+        raise DecompositionUndefinedError
+
+    @staticmethod
+    def compute_matrix(
+        U, control_wires, u_wires, control_values=None
+    ):  # pylint: disable=arguments-differ
+        r"""Representation of the operator as a canonical matrix in the computational basis (static method).
+
+        The canonical matrix is the textbook matrix representation that does not consider wires.
+        Implicitly, this assumes that the wires of the operator correspond to the global wire order.
+
+        Args:
+            U (tensor_like): unitary matrix
+            control_wires (Iterable): the control wire(s)
+            u_wires (Iterable): the wire(s) the unitary acts on
```

Comment:
Oh I see, has the argument been renamed for the controlled operations from `wires` to `u_wires`?

Reply:
For this function, the parameter is called `u_wires` because the parameters for `compute_matrix` use the same names as the operation parameters/hyperparameters.